libs/component-library/src/lib/radio-button/RadioButton.tsx

```diff
@@ -168,34 +154,43 @@ const RadioButton = ({
                 onPress,
                 onValueChange,
                 value,
+                contextValue,
               });
             }
       }
     >
-      <Animated.View
+      <FxRadioBase
+        variant={type}
         style={[
-          s.radio,
+          isMultiSelect ? s.checkbox : s.radio,
           {
-            backgroundColor: radioBackgroundColor,
-            borderColor: radioColor,
             borderWidth: borderAnim,
           },
         ]}
       >
-        {checked ? (
-          <View style={[StyleSheet.absoluteFill, s.radioContainer]}>
-            <Animated.View
-              style={[
-                s.dot,
-                {
-                  backgroundColor: disabled ? colors.border : colors.white,
-                  transform: [{ scale: radioAnim }],
-                },
-              ]}
+        <Animated.View
+          style={[
+            StyleSheet.absoluteFill,
+            s.radioContainer,
+            {
+              transform: [{ scale: radioAnim }],
+            },
+          ]}
+        >
+          {isMultiSelect ? (
+            <FxCheckIcon
+              color={
+                theme.radioCheckmarkVariants[type || 'defaults']
+                  .backgroundColor as keyof typeof theme.colors
```

Comment:
Been digging into this override, it's giving a type error because `color` is a literal type. Literal types are sub-types of a collective type. In this case `string`, hence the error. It would be better to type narrow `radioCheckmarkVariants` than override it with `as` here. 

To do this:  

```diff
theme.radioCheckmarkVariants[type || 'defaults']
-  .backgroundColor as keyof typeof theme.colors
+  .backgroundColor
}
```

Then in theme file:
```diff
const radioTheme = {
  container: {
    types: {
      defaults: {
        borderColor: 'border',
      },
      disabled: {
        backgroundColor: 'backgroundSecondary',
        borderColor: 'border',
      },
      pressed: {
        borderColor: 'greenBase',
      },
      pressedDisabled: {
        borderColor: 'greenBase',
        opacity: 0.5,
      },
    },
  },
  checkmark: {
    types: {
      defaults: {
        backgroundColor: 'transparent',
      },
      disabled: {
        backgroundColor: 'transparent',
      },
      pressed: {
        backgroundColor: 'backgroundApp',
      },
      pressedDisabled: {
        backgroundColor: 'backgroundApp',
      },
    },
  },
-};
+} as const;
```

Reply:
Unfortunately, we cannot utilize this method as it throws an error for lines [495-510](https://github.com/functionland/apps-monorepo/pull/227/files#diff-6e3389ff5714a9cb63c25bc5e1c71d19a36bd6deb4f04993b52523d3383a3588R495-R510) since casting `radioTheme` as `const` makes it readonly 😞